### PR TITLE
refactor: use guild-wide route for fetching active threads

### PR DESCRIPTION
**Pull request by ckohen**
**Please describe the changes this PR makes and why it should be merged:**

Guild wide route for fetching active threads! And since channel wide route its being removed in v10, already based the per channel fetch on guild wide.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: src/managers/GuildChannelManager.js [hunk 1]
```diff
@@ -177,6 +178,27 @@ class GuildChannelManager extends CachedManager {
...
+   * @returns {Promise<FetchedThreads>|Promise<APIActiveThreadsList>}
+   * @example
+   * // Fetch all threads from the guild
+   * message.guild.channels.fetchActiveThreads()
+   *   .then(fetched => console.log(`There are ${fetched.threads.size} threads.`))
+   *   .catch(console.error);
+   */
+  async fetchActiveThreads(cache = true, returnRaw = false) {
+    const raw = await this.client.api.guilds(this.guild.id).threads.active.get();
+    return returnRaw ? raw : ThreadManager._mapThreads(raw, this.client, { guild: this.guild, cache });
```
Comment (SpaceEEC) [inline src/managers/GuildChannelManager.js:195]: This (`returnRaw`) feels a bit out of place, we don't do this anywhere else, do we?

File: typings/index.d.ts [hunk 2]
```diff
@@ -2292,6 +2293,8 @@ export class GuildChannelManager extends CachedManager<
   ): Promise<
     Collection<Snowflake, TextChannel | VoiceChannel | CategoryChannel | NewsChannel | StoreChannel | StageChannel>
   >;
+  public fetchActiveThreads(cache?: boolean): Promise<FetchedThreads>;
+  public fetchActiveThreads(cache: boolean, returnRaw: true): Promise<APIThreadList>;
```
Comment (NotSugden) [inline typings/index.d.ts:2297]: `cache` has no effect if `returnRaw` is true - perhaps change to a single object parameter and omit cache on this overload?

**Review by NotSugden — CHANGES_REQUESTED** [review]

File: src/managers/GuildChannelManager.js (hunk 1 above)
Comment (NotSugden) [reply src/managers/GuildChannelManager.js:195]: what would be the use case for returning the raw threads anyway?

Comment (ckohen) [reply src/managers/GuildChannelManager.js:195]: its for internals, see
https://github.com/discordjs/discord.js/blob/150a295224ea0de5ea541d624addc8489e00b44a/src/managers/ThreadManager.js#L232

Comment (SpaceEEC) [reply src/managers/GuildChannelManager.js:195]: I don't think that should be part of the public API then. :eyes:
Maybe move that to a private method or duplicate that api call into the ThreadManager?

Comment (ckohen) [reply src/managers/GuildChannelManager.js:195]: I was thinking about duplicating the call, especially since its a one-liner. Is that better?

Comment (SpaceEEC) [reply src/managers/GuildChannelManager.js:195]: I think so.

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**